integration/monitor/all-scenarios.js

```diff
@@ -1,12 +1,38 @@
 const cron = require("node-cron");
 const fs = require("fs");
 const path = require("path");
+const Sentry = require("@sentry/node");
 const { Worker } = require("worker_threads");
 
 const { fileToMetricName } = require("../utils");
+const RateLimiter = require("./rate-limiter");
 
 const filesPath = path.join(__dirname, "..", "scenarios");
 
+/*
+ * Create a rate limiter that allows only one lease per hour.
+ */
+const rateLimiter = new RateLimiter(60 * 60 * 1000, 1);
+
+/*
+ * Initialize Sentry with the DSN and environment from the environment variables.
+ * Add a beforeSend callback that will only send events if the rate limiter allows it.
+ */
+Sentry.init({
+  dsn: process.env.SENTRY_DSN,
+  environment: process.env.SENTRY_ENVIRONMENT,
+  beforeSend(event) {
+    if (process.env.SENTRY_ENVIRONMENT == "prod") {
+      return rateLimiter.lease() ? event : null;
```

Comment:
📓 Returning null in this callback does not send the event to Sentry.

https://docs.sentry.io/platforms/node/configuration/filtering/

Reply:
Its a nit-pick, but the ternary (`?`) could be in the `if` above so this line doesn't need to also return `null` like the default return below